overview/cli.md

```diff
@@ -0,0 +1,385 @@
+
+Let’s focus on our first application. We will add colors to the application’s banner. We will use white font on a black background:
+
+```crystal
+app_banner = "#{"The Beatles".colorize(:white).on(:black)} App"
+```
+
+Great! That was easy!
+
+And for our second application, we will add a text decoration (`blink`in this case):
+
+```crystal
+# file: let_it_cli.cr
+require "colorize"
+
+puts "Welcome to The Beatles Sing Along version 1.0!"
+puts "Enter a phrase you want The Beatles to sing"
+print "> "
+user_input = gets
+
+exit if user_input.nil? # Ctrl+D
+
+default_lyrics = "Na, na, na, na-na-na na" \
+                 " / " \
+                 "Na-na-na na, hey Jude"
+
+lyrics = user_input.presence? || default_lyrics
+
+puts "The Beatles are singing: #{"🎵#{user_input}🎶🎸🥁".colorize.mode(:blink)}"
+```
+
+Let’s try the renewed application … and _hear_ the difference!!
+
+And **now** we have two fabulous apps!!
+
+You may read more about the `Colorize` module  [in the official API documentation](https://crystal-lang.org/api/latest/Colorize.html) and find a list of available colors and text decorations.
+
+## Testing
+
+As with any other application, at some point we would like to [write tests](https://crystal-lang.org/reference/guides/testing.html) for the different features.
+
+Right now the code containing the logic of each of the applications always gets executed with the `OptionParser`, i.e. there is no way to include that file without running the whole application. So first we would need to refactor the code, separating the code necessary for parsing options from the logic. Ones the refactor is done, then we could start testing the logic, and so including the file with the logic in the testing files we need. We leave this as an exercise for the reader.
+
+## Using `Readline` and `NCurses`
+
+In case we want to build richer CLI applications, there are libraries that can help us. Here we will name two well-known libraries: `Readline` and` NCurses`.
+
+As stated in the documentation for the [GNU Readline Library](http://www.gnu.org/software/readline/), `Readline` is a library that provides a set of functions for use by applications that allow users to edit command lines as they are typed in.
+`Readline` has some great features: filename autocompletion out of the box; custom autocompletion method; keybinding, just to mention a few. If we want to try it then the [crystal-lang/crystal-readline](https://github.com/crystal-lang/crystal-readline) shard will give us an easy API to use `Readline`.
+
+On the other hand, we have `NCurses`(New Curses). This library allows developers to create _graphical_ user interfaces in the terminal. As its name implies, it is an improved version of the library named `Curses`, which was developed to support a text-based dungeon-crawling adventure game called Rogue!
+As you can imagine, there are already a couple of shards in the ecosystem that will allow us to use `NCurses` in Crystal!
```

Comment:
Would be nice to link to those shards if they're already mentioned.

Reply:
yep, I totally agree! but i didn't test any of those libraries ... yet :), so i didn't want to mention any in particular.

Reply:
Btw, thanks for all the other suggestions!

Reply:
Thanks for the writing! =)

I suppose it would be fine to link available shards and mention that they're not tested. But at least the interested reader gets a direct reference where to look. IMO that's better than just vaguely mentioning "there are some shards (somewhere in the interweb)".

Reply:
Or at least a link to https://crystalshards.org/?filter=ncurses where you can discover these shards.